### PR TITLE
doc: Add mclock configuration reference documentation

### PR DESCRIPTION
**Pull request by sseshasa**
Add the mclock related configuration reference documentation.
The content includes,
- background about mclock profiles
- steps to benchmark OSDs
- steps to enable a mclock profile
- mclock configuration options

Signed-off-by: Sridhar Seshasayee <sseshasa@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sseshasa** [comment]
@jdurgin @neha-ojha Please add the documentation reviewers as appropriate. Thanks!

File: doc/rados/configuration/mclock-config-ref.rst [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+==============================
+
+The following sections outline the steps required to enable a mclock profile:
+
+Determine OSD Capacity Using Benchmark Tests
+--------------------------------------------
+
+To allow mclock to fulfill the QoS goals across its clients, the most important
+criteria is to have a good understanding of each OSD capacity in terms of their
+baseline throughputs (IOPS) across the Ceph nodes . To determine the capacity,
```
Comment (liewegas) [inline doc/rados/configuration/mclock-config-ref.rst:148]: s/nodes ./nodes./

Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst]: Fixed.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+========================
+
+.. index:: mclock; configuration
+
+This dmclock related configuration parameters can be applied using mclock
+profiles. The purpose of the mclock profiles is to mask the low level details
+from a user trying to configure mclock for a specific purpose. At a high level,
+for each OSD in the enviroment the steps involve,
+
+- providing an input related to the total capacity of OSD(s) and
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:12]: we could add a reference to this section https://ceph--40531.org.readthedocs.build/en/40531/rados/configuration/mclock-config-ref/#determine-osd-capacity-using-benchmark-tests, to help the reader understand what capacity means here

File: doc/rados/configuration/mclock-config-ref.rst [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+:Bluestore Throttle Parameters:
+  The recommendation is to use the default values as defined by
+  ``bluestore_throttle_bytes`` and ``bluestore_throttle_deferred_bytes``. But
+  these parameters may also be determined during the benchmarking phase as
+  described below.
+
+Benchmarking Test Steps Using CBT
+`````````````````````````````````
+
+The steps below uses the default shards and details the steps to determine the
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:176]: s/uses/use/

Comment (neha-ojha) [reply doc/rados/configuration/mclock-config-ref.rst:176]: the default number of shards

File: doc/rados/configuration/mclock-config-ref.rst [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+2. Install cbt and all the dependencies mentioned on the cbt github page.
+3. Construct the Ceph configuration file and the cbt yaml file.
+4. Ensure that the bluestore throttle options ( i.e.
+   ``bluestore_throttle_bytes`` and ``bluestore_throttle_deferred_bytes``) are
+   set to the default values.
+5. Ensure that the test is performed on similar device types to get reliable
+   OSD capacity data.
+6. The OSDs can be grouped together with the desired replication factor for the
+   test to ensure reliability of OSD capacity data.
+7. After ensuring that the OSDs nodes are in the desired configuration, run a
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:191]: s/OSDs/OSD/ have the desired configuration

File: doc/rados/configuration/mclock-config-ref.rst [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+   ``bluestore_throttle_deferred_bytes`` to 32 KiB(32768 Bytes) each to begin
+   with. Otherwise, you may skip to the next section.
+10. Run the 4KiB random write workload as before on the OSD(s) for 300 secs.
+11. Note the overall throughput from the cbt log files and compare the value
+    against the baseline throughput in step 8.
+12. If the throughput doesn't match with the baseline, increment the bluestore
+    throttle options by 2x and repeat steps 9 through 11 until the obtained
+    throughput is very close to the baseline value.
+
+For e.g., during benchmarking on a machine with NVMe SSDs a value of 256 KiB for
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:207]: How about "In our benchmarking..."?

**Review by neha-ojha — COMMENTED** [review]
content looks great! left a few nits, I will let @zdover23 proofread once.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 6]
```diff
@@ -0,0 +1,361 @@
+========================
+ mClock Config Reference
+========================
+
+.. index:: mclock; configuration
+
+This dmclock related configuration parameters can be applied using mclock
+profiles. The purpose of the mclock profiles is to mask the low level details
+from a user trying to configure mclock for a specific purpose. At a high level,
+for each OSD in the enviroment the steps involve,
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:10]: also: 
s/involve, /involve/

(remove the comma)

File: doc/rados/configuration/mclock-config-ref.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:207]: @neha-ojha: I suggest that we merge this as it is, and I will make a full edit of it on your Friday and send the edited text to you for approval by the end of your Friday.

**Review by zdover23 — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

File: doc/rados/configuration/mclock-config-ref.rst (hunk 5 above)
Comment (neha-ojha) [reply doc/rados/configuration/mclock-config-ref.rst:207]: sure, we can address my comments in follow-up PR

**Pull request merged**